homeassistant/components/camera/proxy.py

```diff
@@ -0,0 +1,264 @@
+"""
+Proxy camera platform that enables image processing of camera data.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/proxy
+"""
+import logging
+import asyncio
+import aiohttp
+import async_timeout
+
+import voluptuous as vol
+
+from homeassistant.util.async import run_coroutine_threadsafe
+from homeassistant.helpers import config_validation as cv
+import homeassistant.util.dt as dt_util
+from homeassistant.const import (
+    CONF_NAME, CONF_ENTITY_ID, HTTP_HEADER_HA_AUTH)
+from homeassistant.components.camera import (
+    PLATFORM_SCHEMA, Camera)
+from homeassistant.helpers.aiohttp_client import (
+    async_get_clientsession, async_aiohttp_proxy_web)
+
+REQUIREMENTS = ['pillow==5.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MAX_IMAGE_WIDTH = "max_image_width"
+CONF_IMAGE_QUALITY = "image_quality"
+CONF_IMAGE_REFRESH_RATE = "image_refresh_rate"
+CONF_FORCE_RESIZE = "force_resize"
+CONF_MAX_STREAM_WIDTH = "max_stream_width"
+CONF_STREAM_QUALITY = "stream_quality"
+CONF_CACHE_IMAGES = "cache_images"
+
+DEFAULT_BASENAME = "Camera Proxy"
+DEFAULT_QUALITY = 75
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_MAX_IMAGE_WIDTH): int,
+    vol.Optional(CONF_IMAGE_QUALITY): int,
+    vol.Optional(CONF_IMAGE_REFRESH_RATE): float,
+    vol.Optional(CONF_FORCE_RESIZE, False): cv.boolean,
+    vol.Optional(CONF_CACHE_IMAGES, False): cv.boolean,
+    vol.Optional(CONF_MAX_STREAM_WIDTH): int,
+    vol.Optional(CONF_STREAM_QUALITY): int,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
```

Comment:
We've switched to async/await syntax so you can drop `@asyncio.coroutine` and make it `async def async_setup_platform(…)`